src/plugin-client.ts

```diff
@@ -11,26 +11,47 @@
 
 'use strict';
 
-const AddonManagerProxy = require('./addon-manager-proxy');
-const {MessageType} = require('./constants');
-const Deferred = require('./deferred');
-const EventEmitter = require('events');
-const IpcSocket = require('./ipc');
-
-class PluginClient extends EventEmitter {
-  constructor(pluginId, {verbose} = {}) {
+import { AddonManagerProxy } from './addon-manager-proxy';
+import { MessageType } from './constants';
+import { Deferred } from './deferred';
+import { EventEmitter } from 'events';
+import { IpcSocket } from './ipc';
+import WebSocket from 'ws';
+
+export class PluginClient extends EventEmitter {
+  private verbose: boolean;
+  private deferredReply?: Deferred<AddonManagerProxy, void>;
+  private logPrefix: string;
+  private gatewayVersion?: string;
+  private userProfile: any;
```

Comment:
Schema is defined in `schema/messages/plugin-register-response.json → userProfile`. Maybe add it here as well? The same goes for `preferences` one line below.